src/compiler/program.ts

```diff
@@ -1551,6 +1566,30 @@ namespace ts {
             }
         }
 
+        function processLibReferenceDirectives(file: SourceFile) {
+            const libDirectory = getLibDirectory();
+            forEach(file.libReferenceDirectives, libReference => {
+                const libName = libReference.fileName.toLocaleLowerCase();
+                const libFileName = libMap.get(libName);
+                if (libFileName) {
+                    // we ignore any 'no-default-lib' reference set on this import.
+                    processRootFile(combinePaths(libDirectory, libFileName), /*isDefaultLib*/ true, /*ignoreNoDefaultLib*/ false);

```

Comment:
shouldn't be `ignoreNoDefaultLib` be true in order to ignore `'no-default-lib'` references?